### PR TITLE
Adding Shopify detector

### PR DESCRIPTION
**Pull request by kstilwell**
Specifically, this is adding Shopify access token detection. These tokens are used to authenticate with the Shopify admin API.

Some relevant links:
https://shopify.dev/apps/auth/admin-app-access-tokens
https://shopify.dev/api/admin/getting-started

### CODE REVIEW

**Review by dustin-decker — CHANGES_REQUESTED** [review]
Thanks for the contribution, @kstilwell!

If we use this endpoint for verification it should verify for any oauth scopes assigned to the app: https://shopify.dev/api/admin-rest/2022-10/resources/accessscope#get-admin-oauth-access-scopes

We could also use the response data to add extra information on the scopes that the credential has in the `ExtraData` field.

**Comment by kstilwell** [comment]
Hey @dustin-decker, thanks for the feedback...made the changes you requested.

**Review by dustin-decker — APPROVED** [review]
Thanks Kevin, looks good now. We'll merge this soon after updating the protos.

**Comment by kstilwell** [comment]
Hey @dustin-decker, just wanted to check if there was anything else you needed from me to get this merged.

**Comment by ahrav** [comment]
> Hey @dustin-decker, just wanted to check if there was anything else you needed from me to get this merged.

Hey Kevin, looks like there is one linter warning in the detector file. I believe you should be able to omit the `fmt.Sprintf` since `key` is already a string. I can get it merged after that.

File: pkg/detectors/shopify/shopify.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+				Redacted:     domainRes,
+				Raw:          []byte(key + domainRes),
+			}
+
+			if verify {
+				req, err := http.NewRequestWithContext(ctx, "GET", "https://"+domainRes+"/admin/oauth/access_scopes.json", nil)
+				if err != nil {
+					continue
+				}
+				req.Header.Add("X-Shopify-Access-Token", fmt.Sprintf("%s", key))
```
Comment (ahrav) [inline pkg/detectors/shopify/shopify.go:59]: linter suggestion: We can omit the `fmt.Sprintf` here.

**Pull request merged**